### ISSUE TITLE
Refreshing the page shouldn't delete current work.

### ISSUE THREAD

**Issue by simonv3** [jdittrich/quickMockup#32]
If you refresh the page for whatever reason, you shouldn't lose the current work. Can it be stored in localstorage or in the user's cache?

**Comment by jdittrich** [comment]
I sympathize with the idea but as I currently see it, it would mean one would need to implement a "files"-in-localstorage-management-part of the app (or a save-to-url etc), which would sort of double the saving-as-files-part (which I deliberatly choose as means of storing data)

You get the warning, when you reload, don't you? (I know this is not ideal, but that best for now)

**Comment by simonv3** [comment]
Yeah, totally get the warning, and also get the complexity factor. If we switch to an MVC then an abstraction of the objects, as a nested json data object or something similar, which would be used to generate the relevant DOM would also work?

**Comment by paulproteus** [comment]
FWIW, if quickMockup is ported to Sandstorm, I imagine that people are
going to randomly reload it frequently for any of a number of reasons, like
their Internet connection breaking and them wondering if Sandstorm is stuck
and so they reload the page. I know I do that frequently.

Having said that, if quickMockup on Sandstorm autosaves to disk on the
server side, then that addresses this problem.

**Comment by simonv3** [comment]
I think a higher level question of saving to disk is - is this a webapp or a standalone app that just happens to run in the browser?

**Comment by jdittrich** [comment]
> webapp or a standalone app....

It is a standalone app that happens to run in the browser. Or, to be more specific, it is an [unhosted app](http://unhosted.org/). Which means it still can get save-to-localstorage. 

> their Internet connection breaking and them wondering if Sandstorm is stuck

Interesting – it should not be possible that quickMockup is slow due to the connection, since everything is on the client side. (But naturally, they may still reload)
